src/Microsoft.ML.StandardTrainers/Standard/ModelStatistics.cs

```diff
@@ -16,249 +16,345 @@
 using Microsoft.ML.Trainers;
 
 // This is for deserialization from a model repository.
-[assembly: LoadableClass(typeof(LinearModelStatistics), null, typeof(SignatureLoadModel),
+[assembly: LoadableClass(typeof(ModelStatisticsBase), typeof(LinearModelParameterStatistics), null, typeof(SignatureLoadModel),
     "Linear Model Statistics",
-    LinearModelStatistics.LoaderSignature)]
+    LinearModelParameterStatistics.LoaderSignature)]
+
+// This is for deserialization from a model repository.
+[assembly: LoadableClass(typeof(ModelStatisticsBase), typeof(ModelStatisticsBase), null, typeof(SignatureLoadModel),
+    "Model Statistics",
+   ModelStatisticsBase.LoaderSignature)]
 
 namespace Microsoft.ML.Trainers
 {
     /// <summary>
     /// Represents a coefficient statistics object.
     /// </summary>
-    public readonly struct CoefficientStatistics
+    public sealed class CoefficientStatistics
     {
-        public readonly string Name;
         public readonly float Estimate;
         public readonly float StandardError;
         public readonly float ZScore;
         public readonly float PValue;
+        public readonly int FeatureIndex;
 
-        internal CoefficientStatistics(string name, float estimate, float stdError, float zScore, float pValue)
+        internal CoefficientStatistics(int featureIndex, float estimate, float stdError, float zScore, float pValue)
         {
-            Contracts.AssertNonEmpty(name);
-            Name = name;
+            FeatureIndex = featureIndex;
             Estimate = estimate;
             StandardError = stdError;
             ZScore = zScore;
             PValue = pValue;
         }
     }
 
-    // REVIEW: Make this class a loadable class and implement ICanSaveModel.
     // REVIEW: Reconcile with the stats in OLS learner.
     /// <summary>
     /// The statistics for linear predictor.
```

Comment:
>statistics for linear predictor [](start = 12, length = 31)

Is `ModelStatisticsBase` linear model specific?

Reply:
Yes, and LR, MLR are the only classes that set it. You got me thinking: I should derive another class from LinearModelParams and have the stats object only there. LRModelParams. 
I'll log an issue and do that on a separate PR. 

---
In reply to: [266648417](https://github.com/dotnet/machinelearning/pull/2048#discussion_r266648417) [](ancestors = 266648417)